src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -110,6 +110,8 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
                 },
                 _joinableFactory);
 
+            _solutionLoadedEvent = new AsyncManualResetEvent(initialState: true);
```

Comment:
You're right. The perf doesn't matter here in case of a single instance constructor. I'd rather lean toward readability to avoid raising some eyebrows later. Will fix it up.